rfcs/text/0000-create-file-macho.md

```diff
@@ -0,0 +1,152 @@
+
+- Stage: **1 (draft)**
+- Date: **TBD**
+
+Create the Mach Object (Mach-O) sub-field, of the `file` or `process` top-level fieldsets. This document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create the Mach-O sub-field within the `file.` fieldset. This will include 35 sub-fields. `macho` itself is a nested
+field to account for [multiarchitecture binaries](https://en.wikipedia.org/wiki/Fat_binary). Each architecture will be represented
+by a nested objected located at `file.macho`.
+
+| Name                           | Type    | Description                                                     |
+|--------------------------------|---------|-----------------------------------------------------------------|
+| macho.cpu                      | object  | CPU information for the file.                                   |
+| macho.cpu.architecture         | keyword | CPU architecture target for the file.                           |
+| macho.cpu.byte_order           | keyword | CPU byte order for the file.                                    |
+| macho.cpu.subtype              | keyword | CPU subtype for the file.                                       |
+| macho.cpu.type                 | keyword | CPU type for the file.                                          |
+| macho.headers                  | nested  | Header information for the file.                                |
+| macho.headers.commands.number  | long    | Number of load commands for the Mach-O header.                  |
+| macho.headers.commands.size    | long    | Size of load commands of the Mach-O header.                     |
+| macho.headers.commands.type    | keyword | Type of the load commands for the Mach-O header.                |
+| macho.headers.magic            | keyword | Magic field of the Mach-O header.                               |
+| macho.headers.flags            | keyword | Flags set in the Mach-O header.                                 |
+| macho.segments                 | nested  | Segment information for the file.                               |
+| macho.segments.name            | keyword | Name of this segment.                                           |
+| macho.segments.physical_offset | long    | File offset of this segment.                                    |
+| macho.segments.physical_size   | keyword | Amount of memory to map from the file.                          |
+| macho.segments.virtual_address | keyword | Memory address of this segment.                                 |
+| macho.segments.virtual_size    | keyword | Memory size of this segment.                                    |
+| macho.segments.sections        | keyword | Section names contained in this segment.                        |
+| macho.sections                 | nested  | Section information for the segment of the file.                |
```

Comment:
Should this be `macho.segments.sections`? And likewise for the rest of the `macho.sections.*` fields?

Reply:
No, while there is definitely a relationship between segments and sections, they are actually two different views into the binary. Keeping sections and segments at the top-level of the binary structure allows for a more consistent analysis experience across PE, ELF, and MachO